src/wallet/rpc/backup.cpp

```diff
@@ -1879,27 +1879,19 @@ RPCHelpMan restorewallet()
 
     auto backup_file = fs::u8path(request.params[1].get_str());
 
-    if (!fs::exists(backup_file)) {
-        throw JSONRPCError(RPC_INVALID_PARAMETER, "Backup file does not exist");
-    }
-
     std::string wallet_name = request.params[0].get_str();
 
-    const fs::path wallet_path = fsbridge::AbsPathJoin(GetWalletDir(), fs::u8path(wallet_name));
-
-    if (fs::exists(wallet_path)) {
-        throw JSONRPCError(RPC_INVALID_PARAMETER, "Wallet name already exists.");
-    }
-
-    if (!TryCreateDirectories(wallet_path)) {
-        throw JSONRPCError(RPC_WALLET_ERROR, strprintf("Failed to create database path '%s'. Database already exists.", wallet_path.u8string()));
-    }
+    std::optional<bool> load_on_start = request.params[2].isNull() ? std::nullopt : std::optional<bool>(request.params[2].get_bool());
 
-    auto wallet_file = wallet_path / "wallet.dat";
+    DatabaseOptions options;
+    DatabaseStatus status;
+    options.require_existing = true;
+    bilingual_str error;
+    std::vector<bilingual_str> warnings;
 
-    fs::copy_file(backup_file, wallet_file, fs::copy_option::fail_if_exists);
+    const std::shared_ptr<CWallet> wallet = RestoreWallet(context, fs::PathToString(backup_file), wallet_name, load_on_start, options, status, error, warnings);
 
-    auto [wallet, warnings] = LoadWalletHelper(context, request.params[2], wallet_name);
+    HandleWalletError(wallet, status, error);
```

Comment:
In 33763d02ef1011ae71143c5e6e313c72d9b790a6 "refactor: Move restorewallet() logic to the wallet section"

`HandleWalletError` does not handle `FAILED_BAD_PATH` or `FAILED_ALREADY_EXISTS`, which results in a slightly incorrect `RPC_WALLET_ERROR` code. There should be additional error handling for those cases to return better error codes.

Reply:
Done in a9f9a83, which introduced additional error handling.